src/mds/Server.cc

```diff
@@ -5722,6 +5722,47 @@ int Server::parse_layout_vxattr(string name, string value, const OSDMap& osdmap,
   return 0;
 }
 
+int64_t Server::return_bytes(string& value_str)
+{
+  if (value_str.empty()) {
+    return -CEPHFS_EINVAL;
+  }
+  if (value_str.front() == '-') {
+    dout(10) << "negative value not permitted" << dendl;
+    return -CEPHFS_EINVAL;
+  }
+  unordered_map<char, int64_t> size_values = {{'k', 1L << 10}, {'m', 1L << 20}, 
+                                             {'g', 1L << 30}, {'t', 1L << 40}};
+  char data_size = tolower(value_str.back());
+  string value = value_str.substr(0, value_str.length() - 1);
+  if (size_values.find(data_size) != size_values.end()) {
+    try {
+      return (boost::lexical_cast<int64_t>(value)) * size_values[data_size];
+    } catch (boost::bad_lexical_cast const&) {
+      if(value.empty()) {
```

Comment:
nit; redundant code from the else block below - can be reduced.

Reply:
It's not redundant code. `if` block here is for value something like `h/j/k`(single char alpha value + char not found in map) while `else` block is for something like `100Z`(valid number but char not found in map).

Reply:
It can be minimized to just one block but because I'm logging everything over here I need this `if-else`

Reply:
right!